groups/api.py

```diff
@@ -0,0 +1,77 @@
+from ninja import Router
+from django.http import JsonResponse
+from multiset.db_utils import execute_query
+from pathlib import Path
+from groups.services import get_group, create_group, update_group, delete_group
+
+from typing import List, Optional
+from .models import Group, GroupSkeleton
+
+router = Router()
+
+@router.get("/", response=List[Group])
+def get_group_handler(request, group_id: Optional[int] = None, user_id: Optional[int] = None):
+    """
+    Gets a group by its id.
+    Args:
+        group_id: the id of the group to be retrieved
+        user_id: the id of the user to get the groups for
+    Returns:
+        a JSON response with the status of the operation and the group or groups retrieved
+    """
+    try:
+        ret = get_group(group_id, user_id)
+        print("HERE", ret)
+        return ret
+    except Exception as e:
+        return JsonResponse({"status": "error", "message": str(e)}, status=400)
+
+
+@router.post("/create")
+def create_group_handler(request, group: GroupSkeleton, user_ids: List[int] = []):
+    """
+    Creates a new group in the database.
+    Args:
+        group: the group to be created along and the user that created it
+    Returns:
+        a JSON response with the status of the operation and the created group id
+    """
+    
+    if len(user_ids) == 0:
+        return JsonResponse({"status": "error", "message": "No users provided for group"}, status=400)
+    try:
+        ret = create_group(group, user_ids)
+        return ret
+    except Exception as e:
+        return JsonResponse({"status": "error", "message": str(e)}, status=400)
+
+@router.put("/update")
+def update_group_handler(request, group: Group):
+    """
+    Updates a group in the database.
+    Args:
+        group: the group to be updated
+    Returns:
+        a JSON response with the status of the operation
+    """
+    try:
+        ret = update_group(group)
```

Comment:
do we need to return ret here? i think update_group returns null if the update is successful, which isn't very helpful to return to frontend

Reply:
You're right, I'll remove it!